File: WorldFeed/Client/src/app/core/Public/Parameters/HelpClient/GetEndpointRateLimitsParameters.ts
import {GetRateLimitsParameters, IGetRateLimitsParameters, RateLimitsSource} from "./GetRateLimitsParameters";

// For more information read: https://developer.twitter.com/en/docs/developer-utilities/rate-limit-status/api-reference/get-application-rate_limit_status
export interface IGetEndpointRateLimitsParameters extends IGetRateLimitsParameters {
  // Url for which you want to get the rate limit
  url: string;
}

export class GetEndpointRateLimitsParameters extends GetRateLimitsParameters implements IGetEndpointRateLimitsParameters {
  constructor(url?: string, from?: RateLimitsSource, source?: IGetEndpointRateLimitsParameters) {
    if (source) {
      super(source);
      this.url = source?.url;
    } else if (url) {
      this.url = url;
      if (from) {
        super.from = from;
      }
    }
  }

  public url: string;
}

// public GetEndpointRateLimitsParameters(string url)
// {
//     Url = url;
// }
//
//
// public GetEndpointRateLimitsParameters(string url, RateLimitsSource from)
// {
//     Url = url;
//     From = from;
// }

// public GetEndpointRateLimitsParameters(IGetEndpointRateLimitsParameters source) : base(source)
// {
//     Url = source?.Url;
// }

File: Sharebook/Server/src/Services/Identity/Sharebook.Identity.API/Application/Parameters/HelpClient/GetEndpointRateLimitsParameters.cs
namespace Sharebook.Identity.API.Application.Parameters.HelpClient
{
    /// <summary>
    /// For more information read : https://developer.twitter.com/en/docs/developer-utilities/rate-limit-status/api-reference/get-application-rate_limit_status
    /// </summary>
    public interface IGetEndpointRateLimitsParameters : IGetRateLimitsParameters
    {
        /// <summary>
        /// Url for which you want to get the rate limit
        /// </summary>
        string Url { get; set; }
    }

    /// <inheritdoc/>
    public class GetEndpointRateLimitsParameters : GetRateLimitsParameters, IGetEndpointRateLimitsParameters
    {
        public GetEndpointRateLimitsParameters(string url)
        {
            this.Url = url;
        }

        public GetEndpointRateLimitsParameters(string url, RateLimitsSource from)
        {
            this.Url = url;
            base.From = from;
        }

        public GetEndpointRateLimitsParameters(IGetEndpointRateLimitsParameters source) : base(source)
        {
            this.Url = source?.Url;
        }

        /// <inheritdoc/>
        public string Url { get; set; }
    }
}